project_forecast_line/tests/test_forecast_line.py

```diff
@@ -603,10 +603,155 @@ def test_task_forecast_lines_consolidated_forecast_overallocation_multiple_tasks
                 forecast1.employee_resource_forecast_line_id,
                 forecast2.employee_resource_forecast_line_id,
             )
-            self.assertEqual(
-                round(
-                    forecast1.employee_resource_forecast_line_id.consolidated_forecast,
-                    5,
-                ),
-                -0.75000,
+            self.assertAlmostEqual(
+                forecast1.employee_resource_forecast_line_id.consolidated_forecast,
+                -0.75,
+            )
+
+    def test_task_forecast_lines_employee_different_roles(self):
+        """
+        Test forecast lines when employee has different roles.
+
+        Employee has 2 forecast_role_id: consultant 75% and project manager 25%,
+        working 8h per day (standard calendar).
+        Create a task with forecast role consultant, with remaining time = 8h
+        and a scheduled period starting and ending on the same day (22/01/2022 for instance).
+        Assign this task to the user.
+
+        Expected: for the user, on 22/01/2022, 3 forecast lines.
+
+        Expected values are the same for this test and
+        test_task_forecast_lines_employee_main_role
+
+        res_model	                forecast_role_id  forecast_hours consolidated_forecast
+        project.task	            consultant	         -8	             1 (in days)
+        hr.employee.forecast.role	consultant	          6	            -0.25 (in days)
+        hr.employee.forecast.role	project manager	      2	            0.25 (in days)
+
+        """
+        self.env["hr.employee.forecast.role"].create(
+            {
+                "employee_id": self.employee_consultant.id,
+                "role_id": self.role_pm.id,
+                "date_start": "2022-01-01",
+                "rate": 25,
+                "sequence": 1,
+            }
+        )
+        consultant_role = self.env["hr.employee.forecast.role"].search(
+            [
+                ("employee_id", "=", self.employee_consultant.id),
+                ("role_id", "=", self.role_consultant.id),
+            ]
+        )
+        consultant_role.rate = 75
```

Comment:
I would move this inside the with freeze_time() block below. If you don't do this you will have the forecast lines for the employee created relative to the test run date, and likely there will be no line for the 14/01/2022 which is officially in the past as of now.